transcom/mymove
# B-20528 return office users for assignment in office queue payloads

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20528)
## [INT PR](https://github.com/transcom/mymove/pull/13302)

## Summary

This work adds data to the queue payloads, for assigning office users to moves in future work. It takes the GBLOC of the currently logged in user, and the role corresponding to the queue, and returns users of the same role within the GBLOC.

### How to test

1. Login to the office app as a Services Counselor
2. Open the network tab and find the response data
3. Run the following SQL on your local db
```
select ou.id, ou.first_name, ou.last_name, to2.gbloc, r.role_name, p.privilege_name from office_users ou 
join users u on u.id = ou.user_id 
join users_roles ur on ur.user_id = u.id
join roles r on r.id = ur.role_id 
join transportation_offices to2 on ou.transportation_office_id = to2.id
left join users_privileges up on u.id = up.user_id 
left join "privileges" p on p.id = up.privilege_id 
where r.role_type = 'services_counselor'
and to2.gbloc = 'KKFA'
and u.active = true
```
inserting your gbloc into the final line
4. Compare the data and make sure they are the same
5. 
![image](https://github.com/user-attachments/assets/3e4d2a85-477b-41b8-9fa0-d41341ecf4e1)
<img width="1063" alt="Screenshot 2024-07-22 at 3 43 59 PM" src="https://github.com/user-attachments/assets/499dbd6f-e39e-4c39-a27b-dc74cdf360a1">
6. Do the same for TOO and TIO
7. Try and break it
